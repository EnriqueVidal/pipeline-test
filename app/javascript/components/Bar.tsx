import * as React from 'react';
import { Pagination } from '../packs/pipelines.td';

interface Props {
 handleMoveBack: React.MouseEventHandler<HTMLButtonElement>;
  handleMoveForward: React.MouseEventHandler<HTMLButtonElement>;
  handleSelectLimit: React.ChangeEventHandler<HTMLSelectElement>;
  pagination: Pagination;
}

const Bar = ({
  handleMoveBack, handleMoveForward, handleSelectLimit, pagination,
}: Props) => {
  const { page, pages, per_page: perPage } = pagination;

  return (
    <div className="floating-bar columns is-mobile">
      <div className="column">
        <button
          className="button is-link is-small"
          disabled={page === 1}
          onClick={handleMoveBack}
          type="button"
        >
          {'<'}
          {' '}
          Back
        </button>
      </div>
      <div className="column">
        <div className="field">
          <div className="control is-expanded">
            <div className="select is-fullwidth is-small">
              <select id="perPage" onChange={handleSelectLimit} value={perPage}>
                <option value={5}>5</option>
                <option value={10}>10</option>
                <option value={15}>15</option>
                <option value={30}>30</option>
              </select>
            </div>
          </div>
        </div>
      </div>
      <div className="column">
        <div className="buttons is-right">
          <button
            className="button is-link is-small"
            disabled={page === pages}
            onClick={handleMoveForward}
            type="button"
          >
            Forward
            {' '}
            {'>'}
          </button>
        </div>
      </div>
    </div>
  );
};

export default Bar;
