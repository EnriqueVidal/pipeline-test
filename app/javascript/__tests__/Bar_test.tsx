import * as React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import Bar from '../components/Bar';

describe('Bar', () => {
  const handleMoveBack = jest.fn();
  const handleMoveForward = jest.fn();
  const handleSelectLimit = jest.fn();
  const pagination = {
    page: 1,
    per_page: 5,
    pages: 2,
    total: 10,
  };


  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <Bar
          handleMoveBack={handleMoveBack}
          handleMoveForward={handleMoveForward}
          handleSelectLimit={handleSelectLimit}
          pagination={pagination}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('mounts', () => {
    const wrapper = mount(
      <Bar
        handleMoveBack={handleMoveBack}
        handleMoveForward={handleMoveForward}
        handleSelectLimit={handleSelectLimit}
        pagination={pagination}
      />,
    );

    expect(wrapper).not.toBeNull();
  });

  it('clicks Forward', () => {
    const wrapper = mount(
      <Bar
        handleMoveBack={handleMoveBack}
        handleMoveForward={handleMoveForward}
        handleSelectLimit={handleSelectLimit}
        pagination={pagination}
      />,
    );

    const forward = wrapper.find('.button').filterWhere((button) => button.text() === 'Forward >');
    forward.simulate('click');

    expect(handleMoveForward).toHaveBeenCalled();
  });

  it('has Back disabled when sitting on the first page', () => {
    const wrapper = mount(
      <Bar
        handleMoveBack={handleMoveBack}
        handleMoveForward={handleMoveForward}
        handleSelectLimit={handleSelectLimit}
        pagination={pagination}
      />,
    );

    const forward = wrapper.find('.button').filterWhere((button) => button.text() === '< Back');
    expect(forward.props().disabled).toBeTruthy();
  });

  it('selects', () => {
    const wrapper = mount(
      <Bar
        handleMoveBack={handleMoveBack}
        handleMoveForward={handleMoveForward}
        handleSelectLimit={handleSelectLimit}
        pagination={pagination}
      />,
    );


    const perPage = wrapper.find('select').first();
    perPage.simulate('change', { target: { value: 5 } });

    expect(handleSelectLimit).toHaveBeenCalled();
  });
});
