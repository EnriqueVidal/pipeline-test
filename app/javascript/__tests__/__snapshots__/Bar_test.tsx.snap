// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar matches snapshot 1`] = `
<div
  className="floating-bar columns is-mobile"
>
  <div
    className="column"
  >
    <button
      className="button is-link is-small"
      disabled={true}
      onClick={[MockFunction]}
      type="button"
    >
      &lt;
       
      Back
    </button>
  </div>
  <div
    className="column"
  >
    <div
      className="field"
    >
      <div
        className="control is-expanded"
      >
        <div
          className="select is-fullwidth is-small"
        >
          <select
            id="perPage"
            onChange={[MockFunction]}
            value={5}
          >
            <option
              value={5}
            >
              5
            </option>
            <option
              value={10}
            >
              10
            </option>
            <option
              value={15}
            >
              15
            </option>
            <option
              value={30}
            >
              30
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="column"
  >
    <div
      className="buttons is-right"
    >
      <button
        className="button is-link is-small"
        disabled={false}
        onClick={[MockFunction]}
        type="button"
      >
        Forward
         
        &gt;
      </button>
    </div>
  </div>
</div>
`;
